sdk/js/src/EventEmitter.ts

```diff
@@ -93,8 +93,9 @@ export class EventEmitter {
                         variable.callback?.call(variable, variable.value)
                     })
                 }
-                this.emit(`${EventNames.VARIABLE_UPDATED}:*`, key, newVariable)
-                this.emit(`${EventNames.VARIABLE_UPDATED}:${key}`, key, newVariable)
+                const finalVariable = newVariable || null 
```

Comment:
does the event get called if you're never called `client.variable` for a given variable, and it is then removed from the config?

I think that's the bug we need to fix.

Reply:
yeah it gets called for any variable that was previously in the config but removed in this update, regardless of whether it's used in code anywhere. is that not what we want?

Reply:
it is, that was just my suspicion of why they were seeing this issue of the event not being triggered at all.

Reply:
do you think it's better to pass back null in this case or a plain object with what we do know, ie `{key: string, isDefaulted: true}`? the downside to that is sometimes it's called with a DVCVariable instance, and other times a plain object

Reply:
sounds good